tests/test_more.py

```diff
@@ -5803,3 +5804,18 @@ def test_basic(self):
 
     def test_empty(self):
         self.assertEqual(dict(mi.join_mappings()), {})
+
+
+class DiscreteFourierTransformTests(TestCase):
+    def test_basic(self):
+        xarr = [1, 2 - 1j, -1j, -1 + 2j]
+        Xarr = [2, -2 - 2j, -2j, 4 + 4j]
+        self.assertTrue(all(map(isclose, mi.dft(xarr), Xarr)))
+        self.assertTrue(all(map(isclose, mi.idft(Xarr), xarr)))
+
+    def test_roundtrip(self):
+        for _ in range(1_000):
+            N = randrange(35)
+            xarr = [complex(random(), random()) for i in range(N)]
+            Xarr = list(mi.dft(xarr))
+            assert all(map(isclose, mi.idft(Xarr), xarr))
```

Comment:
I would spell this out as `cmath.isclose` rather than just `isclose` because the semantics are so different.